<feat> create entity & repository

Mycodeisnotworking opened the pull request:
- alram / favorite / reviewCharger

Mycodeisnotworking committed:
<feat> create entity & repository

 - alram / favorite / reviewCharger

Mycodeisnotworking committed:
<fix> modify userId(not PK) to userNickName

affected files
 - { Alram, Favorite, ReviewCharger}.java

MinwooJ commented on src/main/java/com/kw/evcs/entity/ReviewCharger.java:
```diff
@@ -0,0 +1,54 @@
+public class ReviewCharger {
+	@Id
+	@GeneratedValue(strategy=GenerationType.IDENTITY)
+	@Column(name="review_id")
```
테이블 이름에 맞춰서 `review_charger_id` 로 하는게 어떰?

Mycodeisnotworking replied:
넹

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/Favorite.java:
```diff
@@ -0,0 +1,42 @@
+	
+	@ManyToOne(fetch = FetchType.LAZY)
+	@JoinColumn(name="charger_id")
+	private Charger charger;
```
즐겨찾기를 충전기 말고 충전소로 하는게 낫지 않아?
즐겨찾기랑 리뷰를 충전기를 기준으로 할지 충전소를 기준으로 할지 생각해봐야할듯

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/Favorite.java:
```diff
@@ -0,0 +1,42 @@
+	
+	@ManyToOne(fetch = FetchType.LAZY)
+	@JoinColumn(name="charger_id")
+	private Charger charger;
```
즐겨찾기를 충전기 말고 충전소로 하는게 낫지 않아?
즐겨찾기랑 리뷰를 충전기를 기준으로 할지 충전소를 기준으로 할지 생각해봐야할듯

MinwooJ replied:
즐겨찾기, 리뷰 둘 다 충전소를 기준으로 하는게 더 맞을 것 같아.

Mycodeisnotworking replied:
즐겨찾기 충전기로 해두면 즐찾해둔거 한눈에 사용 가능한지 아닌지 보여주기 편하지않을까 해서 즐찾 충전소로 해놨어 
리뷰는 충전소 기준으로 하는게 맞겠다

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/Alram.java:
```diff
@@ -0,0 +1,44 @@
+	private Long id;
+
+	@Column(length=255)
+	private String userNickName;
```
User 객체를 갖고 있으니까 이건 빼도 될거같아
userNickname이 User 객체에서 userId를 의미하는거지??

MinwooJ replied:
userId 는 밑에 있는데, 이건 뭐지? 회원가입할 때 userNickName 안받는데..?

Mycodeisnotworking replied:
지울까말까 하다 그냥 써서 올렸어
그럼 닉넴은 아예 안만드는걸로?

Korea-subinpark replied:
응 만약 닉네임 만든다고해도 user 테이블에 만들고 외래키로 접근 가능하니까 여기선 필요 없음

Mycodeisnotworking committed:
<fix> PR opinion reflection

 - delete userNickName (Alram.java)
 - column name review_id => review_charger
 - review now reference ChargingStation

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/ReviewCharger.java:
```diff
@@ -0,0 +1,54 @@
+	@Column(length=32)
+	private String userNickName;
+	@Column
+	private LocalDateTime createdAt;
```
여기에 @CreatedDate, @LastModifiedDate 어노테이션 사용하면 자동으로 시간 등록되게 할 수 있어

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/ReviewCharger.java:
```diff
@@ -0,0 +1,54 @@
+
+@NoArgsConstructor
+@Getter
+@Entity
```
@CreatedDate, @LastModifiedDate 어노테이션 사용하기 위해서 여기에는
@EntityListeners(AuditingEntityListener.class) 추가하면됨

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/ReviewCharger.java:
```diff
@@ -0,0 +1,54 @@
+	@Column(columnDefinition="TEXT")
+	private String body;
+	@Column(length=32)
+	private String userNickName;
```
여기 userNickName은 삭제

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/Favorite.java:
```diff
@@ -0,0 +1,42 @@
+	@GeneratedValue(strategy=GenerationType.IDENTITY)
+	@Column(name="favorite_id")
+	private Long id;
+	@Column(length=255)
```
여기도 userNickName 삭제

Korea-subinpark commented on src/main/java/com/kw/evcs/entity/ReviewCharger.java:
```diff
@@ -0,0 +1,54 @@
+	private LocalDateTime modifiedAt;
+	
+	@ManyToOne
+	@JoinColumn(name="chargingStation_id")
```
charging_station_id 이걸로 해야해

Mycodeisnotworking committed:
<fix> error fixes

 - delete userNickName
 - use JpaAuditing for writeDate / modifiedDate
  - maybe need at User
  - add @EnableJpaAuditing for this at EvcsApplication.java
 - follow charging_station_id column at reviewCharger
 - change Favorite foreign key from charger to charging station

MinwooJ commented:
지금 dev branch로 rebase해서 entity랑 repository 를 domain 안으로 옮겨주세요~

Mycodeisnotworking committed:
[refactor] rebase

 - do with MW
